### PR TITLE
Stop using auto

### LINKED ISSUE #6856: "Fusion" approach to changelog generation
Had to look how bids-specification deals with [src/CHANGES.md](https://github.com/bids-standard/bids-specification/blob/HEAD/src/CHANGES.md) generation: coded in https://github.com/bids-standard/bids-specification/blob/HEAD/.circleci/config.yml#L176 . In a nutshell critical difference is that workflow produces/commits/pushes changelog entry as soon as PR is merged.  This way I think it should be feasible to satisfy everyone 

- we can have autogenerated changelog and quick releasing from PRs as we do now
- we can tool it so it could pickup optionally present in the PR description changelog entry (instead of a title) if so desired
- those who strive for better worded changelog can submit PR/change it while it is all still for `Unreleased` version

Cons: 
- extra commit in target branch for every changelog entry addition (since automated -- I do not really care, FWIW could even be a fancy `datalad run` record)
- we might need/want to agree to keep merge record default as the `Merge pull request #PR` so it could be parsed.  Gotcha can happen if someone pushes some merge from another repo.   But since it is ok to change/modify then before release -- could be fixed.

Notes:
- a typical problem with "manual" changelog modifications in PRs is conflicts (two PRs modify conflicting lines in the same CHANGELOG.md). This approach avoids the conflicts since done by bot.
- *race condition*, in case of rapid merges of two PRs , two runs could compete for pushing new record... smart tool, upon forbidden push, would instead of `reset --hard HEAD^ `, `git pull --ff-only`, and redo addition for that PR it was working on, and then try to push again.  I think this way we can make it "race proof"

We would just need a tool to help adding changelog entries with similar to `auto` sectioning awareness.  There might already be one.  Can research more later into this but first thought to check what @datalad/developers think about this.

edit: may be could be a happy marriage with the `scriv` which @mih at all test-driving for https://github.com/datalad/datalad-next but `scriv` entries auto-produced from PR upon merge/pushed unless PR did provide a scriv entry!?  Then `release` workflow would call `scriv` to compile actual CHANGELOG record.

### LINKED ISSUE #6918: Replace changelog generation with integration with scriv
Auto e2big bug keeps striking and now even for relatively small 0.17.3 release. We need a reliable solution. Also @mih and others arn't satisfied with automated collection of changelog entries and would like to be able to curate them before the release. I think we need a solution which would allow for both

datalad-next already used scriv to assemble changelog entries: https://pypi.org/project/scriv/ . But again it is done manually requiring PR author to use scriv locally to master changelog every PR.

I think it would be best to merge both worlds: allow for automation when entries come from PR titles and can be provided custom and could be curated before release.

What is there is GitHub workflow which reacts on PR merges to maint and master and if PR diff lacks changes to changelog.d -  scriv creates a note based on that PR and pushes that change to github. Release workflow will use scriv  collect to actually create changelog for the release. The rest is to be done by auto.

Interested parties could then curate changelog skippers before release to their liking. Lazy overs like me can just rely on automated ones but still capable of inspecting/fixing.

@jwodder do you think this could work?

### LINKED ISSUE #7016: Release workflow did not get correct version for datalad
I have merged #7009 with `release` label, and which should have adjusted release process to use scriv based changelog.  But it seems smth did not work out for tagging so upload failed due to misformed version `0.17.5+11.g5ee218ae5.dirty`

```
2022-09-07T13:49:01.8742387Z Uploading distributions to https://upload.pypi.org/legacy/
2022-09-07T13:49:02.0027719Z Uploading datalad-0.17.5+11.g5ee218ae5.dirty-py3-none-any.whl
2022-09-07T13:49:02.0121746Z [?25l
2022-09-07T13:49:02.1127828Z [2K[35m  0%[0m [90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[0m [32m0.0/1.5 MB[0m • [36m--:--[0m • [31m?[0m
2022-09-07T13:49:02.2155578Z [2K[35m 10%[0m [91m━━━━[0m[90m╺[0m[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[0m [32m0.1/1.5 MB[0m • [36m00:01[0m • [31m11.0 MB/s[0m
2022-09-07T13:49:02.3178494Z [2K[35m100%[0m [90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[0m [32m1.5/1.5 MB[0m • [33m00:00[0m • [31m22.9 MB/s[0m
2022-09-07T13:49:02.4200545Z [2K[35m100%[0m [90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[0m [32m1.5/1.5 MB[0m • [33m00:00[0m • [31m22.9 MB/s[0m
2022-09-07T13:49:02.4870283Z [2K[35m100%[0m [90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[0m [32m1.5/1.5 MB[0m • [33m00:00[0m • [31m22.9 MB/s[0m
2022-09-07T13:49:02.4870983Z [2K[35m100%[0m [90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[0m [32m1.5/1.5 MB[0m • [33m00:00[0m • [31m22.9 MB/s[0m
2022-09-07T13:49:02.4882535Z [?25h[33mWARNING [0m Error during upload. Retry with the --verbose option for more details. 
2022-09-07T13:49:02.4898271Z [31mERROR   [0m HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/        
2022-09-07T13:49:02.4899033Z          '0.17.5+11.g5ee218ae5.dirty' is an invalid value for Version. Error:   
2022-09-07T13:49:02.4899637Z          Can't use PEP 440 local versions. See                                  
2022-09-07T13:49:02.4900276Z          https://packaging.python.org/specifications/core-metadata for more     
2022-09-07T13:49:02.4900772Z          information.                                                           
2022-09-07T13:49:02.5279119Z ##[error]Process completed with exit code 1.
```

edit: workflow run https://github.com/datalad/datalad/runs/8229500959?check_suite_focus=true

### PR DESCRIPTION
**Pull request by jwodder**
Closes #7016.  See https://github.com/jwodder/pr-merge-release-test for a test of the workflow in action.

Closes #6856 , Closes #6918

TODOs:

- [x] comments on released PRs & issues
- [ ] does not overwrite old release notes; see https://github.com/nedbat/scriv/issues/57 -- was tested to work - we just need scriv release

### CODE REVIEW

**Comment by yarikoptic** [comment]
since we need now to wait for `scriv` to allow to work only on the target latest release for github (judging from author speed of reply it would be done in some foreseable future so we don't need to implement it ourselves), would you be able to add a helper for `released` auto plugin functionality, i.e. to annotate issues and PRs with the version it was fixed in?

**Comment by jwodder** [comment]
@yarikoptic Commenting-script added; you can see it in action in the latest PRs in the test repo linked above.

File: tools/ci/release-comment.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+    def __enter__(self) -> ReleaseCommenter:
+        return self
+
+    def __exit__(self, *_exc) -> None:
+        self.session.close()
+
+    @property
+    def release_url(self) -> str:
+        return f"https://github.com/{self.repo_owner}/{self.repo_name}/releases/tag/{self.release_tag}"
```
Comment (yarikoptic) [inline tools/ci/release-comment.py:55]: code looks right to me here but the only demo issue at https://github.com/jwodder/pr-merge-release-test/issues/8#issuecomment-1245613726 seems to have wrong url to release: [0.0.4](https://github.com/%7Bself.repo_owner%7D/%7Bself.repo_name%7D/releases/tag/%7Bself.release_tag%7D) -- may be was fixed up after that issue closed/released?

Comment (jwodder) [reply tools/ci/release-comment.py:55]: Yes, it was fixed in https://github.com/jwodder/pr-merge-release-test/pull/10 (and the comment on that one has a valid release URL).

Comment (yarikoptic) [reply tools/ci/release-comment.py:55]: cool, thanks!

**Comment by yarikoptic** [comment]
ok, now we just need to wait for @nedbat to improve `scriv` (https://github.com/nedbat/scriv/issues/57) and then we would merge/test-drive in datalad. Then IMHO we should centralize all this goodness into a GitHub action so it could be reused by datalad and its extensions at minimum.  I have initiated https://github.com/datalad/release-action repository for that and made you @jwodder an admin for it happen you want to move this goodness there right now. Note that we have https://github.com/datalad/datalad-action already which could provide some hints on how github action could be developed.

**Comment by yarikoptic** [comment]
note: @jwodder, I have updated description to close more issue and become TODOs instead of shortcomings (already addressed)

**Comment by yarikoptic** [comment]
4 @jwodder and FTR, to mitigate the issue of changelog commit pushed from github actions I have added yarikoptic-gitmate user with Write permissions to this repo and added GITMATE_GITHUB_TOKEN so workflow could use that token instead of default one to push.

**Comment by jwodder** [comment]
@yarikoptic Do you want workflows to run on the changelog-creation commit?  I thought you were concerned about running them on *snippet*-creation commits for the `add-changelog-snippet.yml` workflow.

**Comment by yarikoptic** [comment]
Would it be safe to rerun `add-changelog-snippet.yml` workflow upon such a push because label would no longer be assigned and thus that workflow would just do nothing (no infinite recursion of commits to happen)?

**Comment by yarikoptic** [comment]
good observation!  My thinking is:  we can then removed label before pushing the snippet.  But then if snippet push fails - we would end up without the snippet and label, which would be odd... but probably it would be also as odd if now push of snippet fails and we do not have label removed.  So IMHO it would be ok to just swap to steps in the order so we would completely avoid such a race condition.  Am I missing any potential drawback/side-effect @jwodder?

**Comment by jwodder** [comment]
@yarikoptic I can't think of any drawbacks.  Should I update `add-changelog-snippet.yml` in this PR or a separate one?

**Comment by yarikoptic** [comment]
> @yarikoptic I can't think of any drawbacks. Should I update `add-changelog-snippet.yml` in this PR or a separate one?

I think you can do right in this one and we will merge it altogether to give it a shot.

**Comment by jwodder** [comment]
@yarikoptic Steps swapped.  Do you also want `GITMATE_GITHUB_TOKEN` to be used in either or both workflows in this PR?

**Comment by yarikoptic** [comment]
> @yarikoptic Steps swapped. 

Thank you!

> Do you also want `GITMATE_GITHUB_TOKEN` to be used in either or both workflows in this PR?

I don't care much but may be let's do for both for consistency?

**Comment by jwodder** [comment]
@yarikoptic Done: a73411af8.

**Comment by yarikoptic** [comment]
Thank you @jwodder !  So I will proceed with merge and we will try it out

**Comment by yarikoptic** [comment]
FWIW new scriv https://pypi.org/project/scriv/0.17.0/ was released hours back and the change is

The github_release command now only considers the top-most entry in the changelog. You can use the --all option to continue the old behavior of making or updating GitHub releases for all of the entries.

so I do not think we need any tune up to this PR.

**Pull request merged**